relentless/simulate/hoomd.py

```diff
@@ -0,0 +1,784 @@
+
+    """
+    def __init__(self, r_buff=0.4):
+        self.r_buff = r_buff
+
+    def extract_box_params(self, sim):
+        """Extracts HOOMD box parameters (*Lx*, *Ly*, *Lz*, *xy*, *xz*, *yz*)
+        from the simulation's ensemble volume.
+
+        Parameters
+        ----------
+        sim : :py:class:`Simulation`
+            Simulation object.
+
+        Returns
+        -------
+        array_like
+            Array of the simulation box parameters.
+
+        Raises
+        ------
+        ValueError
+            If the volume is not set.
+        TypeError
+            If the volume does not derive from :py:class:`TriclinicBox`.
+
+        """
+        # cast simulation box in HOOMD parameters
+        V = sim.ensemble.V
+        if V is None:
+            raise ValueError('Box volume must be set.')
+        elif not isinstance(V, TriclinicBox):
+            raise TypeError('HOOMD boxes must be derived from TriclinicBox')
+
+        Lx = V.a[0]
+        Ly = V.b[1]
+        Lz = V.c[2]
+        xy = V.b[0]/Ly
+        xz = V.c[0]/Lz
+        yz = V.c[1]/Lz
+
+        return np.array([Lx,Ly,Lz,xy,xz,yz])
+
+    def make_snapshot(self, sim):
+        """Creates a particle snapshot and box for the simulation context.
+
+        Parameters
+        ----------
+        sim : :py:class:`Simulation`
+            Simulation object.
+
+        Returns
+        -------
+        `hoomd.data` snapshot
+            Particle simulation snapshot.
+        :py:class:`freud.Box`
+            Particle simulation box.
+
+        Raises
+        ------
+        ValueError
+            If the particle number is not set.
+
+        """
+        # get total number of particles
+        N = 0
+        for t in sim.ensemble.types:
+            if sim.ensemble.N[t] is None:
+                raise ValueError('Number of particles for type {} must be set.'.format(t))
+            N += sim.ensemble.N[t]
+
+        # cast simulation box in HOOMD parameters
+        Lx,Ly,Lz,xy,xz,yz = self.extract_box_params(sim)
+
+        # make the empty snapshot in the current context
+        with sim.context:
+            box = hoomd.data.boxdim(Lx=Lx,Ly=Ly,Lz=Lz,xy=xy,xz=xz,yz=yz)
+            snap = hoomd.data.make_snapshot(N=N,
+                                            box=box,
+                                            particle_types=list(sim.ensemble.types))
+            # freud boxes are more useful than HOOMD boxes, so prefer that type
+            box = freud.Box.from_box(box)
+
+        return snap,box
+
+    def attach_potentials(self, sim):
+        """Adds tabulated pair potentials to simulation object.
+
+        Parameters
+        ----------
+        sim : :py:class:`Simulation`
+            Simulation object.
+
+        Raises
+        ------
+        ValueError
+            If the length of the *r*, *u*, and *f* arrays are not all equal.
+
+        """
+        # first write all potentials to disk
+        os.chdir(sim.directory.path)
```

Comment:
There is no need to call this explicitly. You can either do:
```py
with sim.directory:
   # in the dir
```
or use `Directory.file()` will return the _absolute_ path so that you don't need to change directory at all.

In this case, we don't need this line at all, so remove.